File: ShapesPainter/app/src/main/java/com/example/alexander/shapespainter/Triangle.java
package com.example.alexander.shapespainter;

import android.graphics.Color;

/**
 * Created by Alexander on 20.01.2017.
 */

public class Triangle extends Shape  {

    public Triangle(Vector2f vertex1,
                    Vector2f vertex2,
                    Vector2f vertex3,
                    Color fillColor,
                    Color outlineColor) {

    }

    @Override
    void draw(ICanvas canvas) {

    }

    @Override
    public ShapeType getType() {
        return null;
    }

    @Override
    public ShapeDiagram getDiagram() {
        return null;
    }

    @Override
    public Color getFillColor() {
        return null;
    }

    @Override
    public Color getOutlineColor() {
        return null;
    }

    @Override
    public Vector2f getVertices() {
        return null;
    }

    @Override
    public boolean isPointInside(Vector2f point) {
        return false;
    }

    @Override
    public Vector2f getCenter() {
        return null;
    }

    @Override
    public Vector2f getSize() {
        return null;
    }
}

File: Fabric/app/src/main/java/com/example/alexander/fabric/DesignerTest/Triangle.kt
package com.example.alexander.fabric.DesignerTest

import com.example.alexander.fabric.Definitions.ColorEnum
import com.example.alexander.fabric.Definitions.CordPoint
import com.example.alexander.fabric.WorkWithShapes.ICanvas
import com.example.alexander.fabric.WorkWithShapes.Shape

/**
 * Created by Alexander on 25.06.2016.
 */
class Triangle(val vertex1: CordPoint, val vertex2: CordPoint, val vertex3: CordPoint, color: ColorEnum): Shape(color)
{
    override fun Draw(canvas: ICanvas)
    {
        canvas.SetColor(color)
        canvas.DrawLine(vertex1, vertex2)
        canvas.DrawLine(vertex2, vertex3)
        canvas.DrawLine(vertex3, vertex1)
    }
    override fun toString(): String {
        return "Triangle: color: ${color.name} data: ${vertex1.toString()},  ${vertex2.toString()},  ${vertex3.toString()}"
    }
}